iothub/service/src/IotHubServiceClient.cs

```diff
@@ -203,6 +209,42 @@ public IotHubServiceClient(string hostName, AzureSasCredential credential, IotHu
         /// </summary>
         /// <seealso href="https://docs.microsoft.com/azure/iot-hub/iot-hub-devguide-messages-c2d"/>.
         public MessagesClient Messages { get; protected set; }
+        
+        /// <summary>
+        /// Requests connection string for the built-in Event Hubs messaging endpoint of the associated IoT hub.
+        /// </summary>
+        /// <param name="cancellationToken">The token which allows the operation to be canceled.</param>
+        /// <returns>A connection string which can be used to connect to the Event Hubs service and interact with the IoT hub messaging endpoint.</returns>
+        /// <exception cref="InvalidOperationException">The Event Hubs host information was not returned by the IoT hub service.</exception>
+        /// <seealso href="https://docs.microsoft.com/azure/iot-hub/iot-hub-devguide-endpoints"/>
+        /// <seealso href="https://docs.microsoft.com/azure/iot-hub/iot-hub-devguide-messages-read-builtin"/>
+        /// <seealso href="https://docs.microsoft.com/azure/iot-hub/iot-hub-amqp-support#receive-telemetry-messages-service-client"/>
+        public Task<string> GetEventHubCompatibleConnectionStringAsync(CancellationToken cancellationToken)
```

Comment:
Expose that functionality in the client, with overloads to account for which ctor someone used.